### DISCUSSION TITLE
Use llm to extract abstract/keywords from articles, example from food safety

### DISCUSSION THREAD

**Discussion by pvgenuchten** [soilwise-he/metadata-augmentation#18] [Ideas]
a sample of a study using llm to extract abstracts/keywords from a set of documents

https://arxiv.org/pdf/2405.15787

suggested by Anna

**Comment by robknapen** [comment]
From the paper: "To prompt the LLM with **relevant scientific abstracts** and correctly **identify the chemical names** in the responses of the LLM, **data collection and preprocessing steps were required for both the set of abstracts and a chemical dictionary**. After filtering the collected abstracts related to a specific food, the LLM is prompted **to extract the mentioned chemical contaminants in the abstracts**. The final steps aim to parse the responses of the LLM to generate the final list of chemical hazards for a food item and evaluate their correctness."

Sure this can be done (although I think their method has a kind off build-in high success rate). But I see it as a different task than extracting an abstract from a document (summarising), or finding the best keywords for a document or abstract (to classify it within a corpus). However we can experiment with all such tasks of course for knowledge management. Having a dictionary/vocabulary/ontology to match key terms or phrases against is usually a good idea.
